### PR TITLE
Add recurringJobs to StorageClass

### LINKED ISSUE #1129: [FEATURE] Set up recurring jobs from the helm chart
**Is your feature request related to a problem? Please describe.**

I would like to install the helm chart enabling recurring jobs 

**Describe the solution you'd like**

Based on how to [Set up recurring jobs using StorageClass](https://github.com/longhorn/longhorn/blob/c91680aa56beedb60c751163194ec5a26177615e/docs/snapshot-backup.md#set-up-recurring-jobs-using-storageclass), be able to use values for `reccurringJobs` parameter from the [chart/templates/storageclass.yaml](https://raw.githubusercontent.com/longhorn/longhorn/master/chart/templates/storageclass.yaml)

**Describe alternatives you've considered**
```yml
parameters:
  numberOfReplicas: "{{ .Values.persistence.defaultClassReplicaCount }}"
  staleReplicaTimeout: "30"
  fromBackup: ""
  baseImage: ""
  {{- if .Values.recurringJobs.enable }}
  recurringJobs: "{{ .Values.recurringJobs.jobsList }}"
  {{- end }}
```

**Additional context**
Same goes for `fromBackup` parameter BTW

### PR DESCRIPTION
**Pull request by jackivanov**
Fixes #1129

### CODE REVIEW

**Comment by yasker** [comment]
We can add an option to the storage class for recurring jobs, but we cannot add a default value to it, since the user should choose how they would want to setup the recurring job. Also, if there is no backup target, we cannot have a recurring backup job as well.

**Comment by jackivanov** [comment]
@yasker I corrected the templates a bit, take a look please

File: chart/templates/storageclass.yaml [hunk 1]
```diff
@@ -12,6 +12,6 @@ parameters:
   staleReplicaTimeout: "30"
   fromBackup: ""
   baseImage: ""
-  {{- if .Values.recurringJobs.enable }}
+  {{- if and .Values.recurringJobs.enable .Values.defaultSettings.backupTarget }}
```
Comment (yasker) [inline chart/templates/storageclass.yaml:15]: In fact we don't need to check the backupTarget since recurring snapshot is a valid choice.

**Comment by yasker** [comment]
Also, please squash the commits into one.

**Pull request merged**